src/player.lua

```diff
@@ -485,7 +491,8 @@ function Player:update(dt, map)
     end
   end
 
-  if halfjumped and self.velocity.y < -450 and not self.rebounding and self.jumping then
+  if halfjumped and self.velocity.y < -450 and not self.rebounding and
+     self.jumping and self:canStand(map)  then
```

Comment:
self.jumping can't be true unless self:canStand returns true as well. So checking both isn't necessary. Can probably just check self.jumping alone.